In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [3]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB


5. How many rows and columns are in `gdp_df`? What are the data types of each column?
-ANSWER: 4 columns and 7662 rows; The data types are float, integer and object.

In [4]:
gdp_df.shape

(7662, 4)

6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [5]:
gdp_df = gdp_df.drop(columns = ['Value Footnotes'])
gdp_df.columns = ['Country', 'Year', 'GDP_Per_ Capita']
gdp_df.head()

Country  Year  GDP_Per_ Capita
0  Afghanistan  2021      1517.016266
1  Afghanistan  2020      1968.341002
2  Afghanistan  2019      2079.921861
3  Afghanistan  2018      2060.698973
4  Afghanistan  2017      2096.093111

7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice? 

In [6]:
 gdp_df['Year'].nunique()

33

In [7]:
gdp_df['Year'].unique()

array([2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014, 2013, 2012, 2011,
       2010, 2009, 2008, 2007, 2006, 2005, 2004, 2003, 2002, 2022, 2001,
       2000, 1999, 1998, 1997, 1996, 1995, 1994, 1993, 1992, 1991, 1990],
      dtype=int64)

8. In this question, you're going to create some plots to show the distribution of GDP per capita for the year 2020. Go to the Python Graph Gallery (https://www.python-graph-gallery.com/) and look at the different types of plots under the Distribution section. Create a histogram, a density plot, a boxplot, and a violin plot. What do you notice when you look at these plots? How do the plots compare and what information can you get out of one type that you can't necessarily get out of the others?

#CREDS TO RENUKA FOR QUESTION 8
sns.violinplot(data = gdp_2020, y = 'GDP_Per_Capita')
plt.ylabel('GDP Per Capita')
plt.title('voilin plot for GDP per capita')
#sns.violinplot(y=gdp_2020['GDP_Per_Capita']) 
plt.show()

In [22]:
gdp_2020 = gdp_df['Year']==2020

In [28]:
sns.violinplot( x = [gdp_2020], y = gdp_df['GDP_Per_Capita'])
plt.ylabel('GDP_Per_Capita'), plt.title('GDP_Per_Capita Violin Plot')

KeyError: 'GDP_Per_Capita'

#Histogram using Seaborn
sns.displot(data=gdp_2020['GDP_Per_Capita'],kde=True)

#Create a density plot using matplotlib
gdp_2020.GDP_Per_Capita.plot.density(color='green')
plt.title('Denisty plot for GDP per capita using matplotlib')
plt.show()

#Create a density plot using seaborn
sns.kdeplot(data = gdp_2020, x= 'GDP_Per_Capita', fill = 'skyblue')
plt.xlabel('GDP Per Capita')
plt.title('Density plot for GDP per capita using seaborn')
plt.show()

# Boxplot using Pandas
gdp_2020.boxplot(column =['GDP_Per_Capita'], grid = False)
#plt.xlabel('GDP Per Capita')
plt.title('Boxplot for GDP per capita')
plt.show()

 # Boxplot using Seaborn
sns.boxplot(data =gdp_2020, x='GDP_Per_Capita')
plt.xlabel('GDP Per Capita')
plt.title('Boxplot for GDP per capita using Seaborn')
plt.show()

9. What was the median GDP per capita value in 2020?

df_2020['GDP_Per_Capita'].median()

10. For this question, you're going to create some visualizations to compare GDP per capita values for the years 1990, 2000, 2010, and 2020. Start by subsetting your data to just these 4 years into a new DataFrame named gdp_decades. Using this, create the following 4 plots:
	* A boxplot
	* A barplot (check out the Barplot with Seaborn section: https://www.python-graph-gallery.com/barplot/#Seaborn)
	* A scatterplot
	* A scatterplot with a trend line overlaid (see this regplot example: https://www.python-graph-gallery.com/42-custom-linear-regression-fit-seaborn)  
Comment on what you observe has happened to GDP values over time and the relative strengths and weaknesses of each type of plot.

In [14]:
gdp_df

Country  Year  GDP_Per_ Capita
0     Afghanistan  2021      1517.016266
1     Afghanistan  2020      1968.341002
2     Afghanistan  2019      2079.921861
3     Afghanistan  2018      2060.698973
4     Afghanistan  2017      2096.093111
...           ...   ...              ...
7657     Zimbabwe  1994      2670.106615
7658     Zimbabwe  1993      2458.783255
7659     Zimbabwe  1992      2468.278257
7660     Zimbabwe  1991      2781.787843
7661     Zimbabwe  1990      2704.757299

[7662 rows x 3 columns]

In [16]:
gdp_df['Year']==1990

0       False
1       False
2       False
3       False
4       False
        ...  
7657    False
7658    False
7659    False
7660    False
7661     True
Name: Year, Length: 7662, dtype: bool

In [17]:
gdp_df['Year'].isin([1990, 2000, 2010, 2020])

0       False
1        True
2       False
3       False
4       False
        ...  
7657    False
7658    False
7659    False
7660    False
7661     True
Name: Year, Length: 7662, dtype: bool

In [ ]:
#RENUKA - gdp_2020 = gdp_df[gdp_df["Year"]==2020]

In [18]:
 gdp_decades=gdp_df[gdp_df['Year'].isin([1990,2000,2010,2020])]

11. Which country was the first to have a GDP per capita greater than $100,000?

TypeError: '>' not supported between instances of 'list' and 'int'